src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1619,21 +1619,41 @@ private void NotifyInvoker(string functionName, Exception ex)
 
         private void OnFileChanged(FileSystemEventArgs e)
         {
-            string directory = GetRelativeDirectory(e.FullPath, ScriptConfig.RootScriptPath);
-            bool isWatchedDirectory = ScriptConfig.WatchDirectories.Contains(directory);
-
             // We will perform a host restart in the following cases:
             // - the file change was under one of the configured watched directories (e.g. node_modules, shared code directories, etc.)
             // - the host.json file was changed
             // - a function.json file was changed
+            // - a proxies.json file was changed
             // - a function directory was added/removed/renamed
             // A full host shutdown is performed when an assembly (.dll, .exe) in a watched directory is modified
+
+            string changeDesciprtioin = string.Empty;
+            string directory = GetRelativeDirectory(e.FullPath, ScriptConfig.RootScriptPath);
             string fileName = Path.GetFileName(e.Name);
-            if (isWatchedDirectory ||
-                (string.Compare(fileName, ScriptConstants.HostMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0 ||
-                 string.Compare(fileName, ScriptConstants.FunctionMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0 ||
-                 string.Compare(fileName, ScriptConstants.ProxyMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0) ||
-                !_directorySnapshot.SequenceEqual(Directory.EnumerateDirectories(ScriptConfig.RootScriptPath)))
+
+            if (ScriptConfig.WatchDirectories.Contains(directory) ||
+                string.Equals("bin", directory, StringComparison.OrdinalIgnoreCase))
```

Comment:
can we just add 'bin' to watched ScriptConfig.WatchDirectories by default? Also, it seems like the 'bin' check is being added to the logic - why?

Reply:
We've added 'bin' to to watched directories automatically in 2.0, consider using the same logic we have there. It would indeed be nice to have a well stated scenario so we understand the goal of this change.